### PR TITLE
Add global random seed

### PR DESCRIPTION
**Pull request by mducducd**
Add a Seed utility class to simplify and unify random seed management across PyTorch, NumPy, and Python’s random. It provides methods to set global seeds and initialize DataLoader workers reproducibly. Now available in advanced_config in config.yaml.

### CODE REVIEW

**Review by EzicStar — CHANGES_REQUESTED** [review]
Good job. I left some minor comments on some modules. To get this merged it is still required to:

- [x] Create a test that covers seeding logic. In which you could for example test that dataloaders work in a reproducible way. Also please check that two dataloaders don't get the exact same generated values. 
- [ ] Update getting-started.md to clarify how reproducibility works, explaining things such as that we don't use [torch.use_deterministic_algorithms()](https://docs.pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html#torch.use_deterministic_algorithms) because of performance drops. Cite [pytorch's reproducibility docs](https://docs.pytorch.org/docs/stable/notes/randomness.html).

**Review by EzicStar — APPROVED** [review]
Good job!
